api/v1alpha1/trafficmanagerbackend_types.go

```diff
@@ -0,0 +1,149 @@
+// +kubebuilder:printcolumn:JSONPath=`.spec.profile.name`,name="Profile",type=string
+// +kubebuilder:printcolumn:JSONPath=`.spec.profile.namespace`,name="Profile-Namespace",type=string
+// +kubebuilder:printcolumn:JSONPath=`.spec.endpointRef.name`,name="Backend",type=string
+// +kubebuilder:printcolumn:JSONPath=`.status.conditions[?(@.type=='Accepted')].status`,name="Is-Accepted",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+
+// TrafficManagerBackend is used to manage the Azure Traffic Manager Endpoints using cloud native way.
+// A backend contains one more endpoints. Therefore, the controller may create multiple endpoints under the Traffic
+// Manager Profile.
+// https://learn.microsoft.com/en-us/azure/traffic-manager/traffic-manager-endpoint-types
+type TrafficManagerBackend struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of TrafficManagerBackend.
+	Spec TrafficManagerBackendSpec `json:"spec"`
+
+	// The observed status of TrafficManagerBackend.
+	// +optional
+	Status TrafficManagerBackendStatus `json:"status,omitempty"`
+}
+
+type TrafficManagerBackendSpec struct {
+	// +required
+	// immutable
+	// Which TrafficManagerProfile the backend should be attached to.
+	Profile ProfileRef `json:"profile"`
+
+	// The reference to a backend.
+	// +required
+	Backend TrafficManagerBackendRef `json:"backend"`
+
+	// The total weight of endpoints behind the serviceImport when using the 'Weighted' traffic routing method.
+	// Possible values are from 1 to 1000.
+	// It is required when the routing method is 'Weighted'.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=1000
+	// For example, if there are two clusters exporting the service via public ip, each public ip will be configured
+	// as "Weight"/2.
+	Weight *int64 `json:"weight,omitempty"`
+}
+
+// ProfileRef is a reference to a trafficManagerProfile object in the same namespace as this TrafficManagerBackend object.
+type ProfileRef struct {
+	// Name is the name of the referenced trafficManagerProfile.
+	Name string `json:"name"`
+}
+
+// TrafficManagerBackendRef is the reference to a backend.
+// Currently, we only support one backend type: ServiceImport.
+type TrafficManagerBackendRef struct {
+	// Name is the reference to the ServiceImport in the same namespace as this TrafficManagerBackend object.
+	// +required
+	Name string `json:"name"`
+}
+
+// TrafficManagerEndpointStatus is the status of Azure Traffic Manager endpoint which is successfully accepted under the traffic
+// manager Profile.
+type TrafficManagerEndpointStatus struct {
+	// Name of the endpoint.
+	// +required
+	Name string `json:"name"`
+
+	// The weight of this endpoint when using the 'Weighted' traffic routing method.
+	// Possible values are from 1 to 1000.
+	// +optional
+	Weight *int64 `json:"weight,omitempty"`
```

Comment:
Is this field gonna be different from the one in `TrafficManagerEndpointSpec`?

Reply:
Do you mean "TrafficManagerBackendSpec"?
The weight in the TrafficManagerBackendSpec is for the backend (ServiceImport): total weight of all the endpoints behind the ServiceImport.

So for each TrafficManagerEndpoint, weight = weight in the spec / number of exported services.

Reply:
oh I see. I was confused with `TrafficManagerBackend*` and `TrafficManagerEndpoint*`